app/helpers/admissions_helper.rb

```diff
@@ -35,4 +35,12 @@ def facet_count_for_gender(value)
   def default_value(value)
     params[value]
   end
+
+  def friendly_date(value)
+    date_string = value.to_s
+    year = date_string[0..3]
+    month = date_string[4..5]
+    day = date_string[6..7]
+    "#{month}/#{day}/#{year}"
+  end
```

Comment:
That’s what I would have expected but it’s a lot more elegant.


--

Sean Crowe
Digital Analyst and Developer Librarian

Digital Projects and Repositories.
University of Cincinnati Libraries
PO Box 210033
Cincinnati OH 45221-0033
Tel: (513) 556-1899
Fax: (513) 556-4393
Email: sean.crowe@uc.edu<mailto:sean.crowe@uc.edu>
gchat: crowesn






On May 5, 2016, at 10:53 PM, James Van Mil <notifications@github.com<mailto:notifications@github.com>> wrote:


In app/helpers/admissions_helper.rb<https://github.com/jamesvanmil/house-of-refuge/pull/87#discussion_r62283757>:

> @@ -35,4 +35,12 @@ def facet_count_for_gender(value)
>    def default_value(value)
>      params[value]
>    end
> +
> +  def friendly_date(value)
> +    date_string = value.to_s
> +    year = date_string[0..3]
> +    month = date_string[4..5]
> +    day = date_string[6..7]
> +    "#{month}/#{day}/#{year}"
> +  end


Okay I tested them (mine is with the regex in the name):

> Benchmark.bm do |x|
>   x.report { 100000.times do; friendly_date(19830101); end }
>   x.report { 100000.times do; friendly_date_regex(19830101); end }
> end
       user     system      total        real
   0.110000   0.000000   0.110000 (  0.114900)
   0.250000   0.000000   0.250000 (  0.261434)

The regex way is slower, an extra .15 seconds when you run it 100000 times!

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/jamesvanmil/house-of-refuge/pull/87/files/923c525e8672694b8d86a5a4def489ab0202c8f2#r62283757>